### ISSUE TITLE
Not saving last checkpoint when max_steps area reached

### ISSUE THREAD

**Issue by anaprietonem** [ecmwf/anemoi-training#183]
### What happened?

When using `max_steps` we can ensure the training loop ends at the same time the optimizer scheduler. However currently we are not saving the 'last' checkpoint unless it matches the end of the epoch of the time frequency specified in the config

### What are the steps to reproduce the bug?

check what checkpoints are saved when:
max_steps do not match exactly with end of epoch.

### Version

Current

### Platform (OS and architecture)

ATOS

### Organisation

ECMWF

**Comment by Rilwan-Adewoyin** [comment]
Happy to look into this now if no one else has already started? @anaprietonem
